File: app/src/main/java/farasense/mobile/view/ui/dialog/AlertGpsDialog.java
package farasense.mobile.view.ui.dialog;

import android.app.Activity;
import android.app.Dialog;
import android.os.Bundle;
import android.widget.Button;

import farasense.mobile.R;
import farasense.mobile.util.GpsUtil;

public class AlertGpsDialog extends Dialog {

    private Activity activity;
    private Button acceptButton;
    private Button exitButton;

    public AlertGpsDialog(Activity activity) {
        super(activity);
        this.activity = activity;
    }

    @Override
    protected void onCreate(Bundle savedInstance){
        super.onCreate(savedInstance);
    }

    @Override
    protected void onStart() {
        super.onStart();

        initDialog();

        acceptButton.setOnClickListener(c -> {
            GpsUtil.turnOnGps(activity);
            this.dismiss();
        });

        exitButton.setOnClickListener(c -> {
            activity.finish();
        });
    }

    private void initDialog() {
        acceptButton = findViewById(R.id.button_accept);
        exitButton = findViewById(R.id.button_exit);
    }
}

File: app/src/main/java/old_farasense/dialog/AlertGpsDialog.kt
//package old_farasense.dialog
//
//import android.app.Activity
//import android.app.Dialog
//import android.widget.Button
//
//import farasense.mobile.R
//import old_farasense.util.GpsUtil
//
//class AlertGpsDialog(private val activity: Activity) : Dialog(activity) {
//    private var acceptButton: Button? = null
//    private var exitButton: Button? = null
//
//    override fun onStart() {
//        super.onStart()
//
//        initDialog()
//
//        acceptButton!!.setOnClickListener { c ->
//            GpsUtil.turnOnGps(activity)
//            this.dismiss()
//        }
//
//        exitButton!!.setOnClickListener { c -> activity.finish() }
//    }
//
//    private fun initDialog() {
//        acceptButton = findViewById(R.id.button_accept)
//        exitButton = findViewById(R.id.button_exit)
//    }
//}
